extensions/mybatis/deployment/src/main/java/org/apache/camel/quarkus/component/mybatis/deployment/MybatisProcessor.java

```diff
@@ -34,13 +39,19 @@ FeatureBuildItem feature() {
         return new FeatureBuildItem(FEATURE);
     }
 
-    /**
-     * Remove this once this extension starts supporting the native mode.
-     */
-    @BuildStep(onlyIf = NativeBuild.class)
-    @Record(value = ExecutionTime.RUNTIME_INIT)
-    void warnJvmInNative(JvmOnlyRecorder recorder) {
-        JvmOnlyRecorder.warnJvmInNative(LOG, FEATURE); // warn at build time
-        recorder.warnJvmInNative(FEATURE); // warn at runtime
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    CamelBeanBuildItem mybatisComponent(
+            List<SqlSessionFactoryBuildItem> sqlSessionFactoryBuildItem, CamelMyBatisRecorder recorder) {
+        return new CamelBeanBuildItem("mybatis", MyBatisComponent.class.getName(),
+                recorder.createMyBatisComponent(sqlSessionFactoryBuildItem.get(0).getSqlSessionFactory()));
+    }
+
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    CamelBeanBuildItem mybatisBeanComponent(
+            List<SqlSessionFactoryBuildItem> sqlSessionFactoryBuildItem, CamelMyBatisRecorder recorder) {
+        return new CamelBeanBuildItem("mybatis-bean", MyBatisBeanComponent.class.getName(),
+                recorder.createMyBatisBeanComponent(sqlSessionFactoryBuildItem.get(0).getSqlSessionFactory()));
```

Comment:
Can we be sure that sqlSessionFactoryBuildItem is not empty?

Reply:
sure, I create https://github.com/quarkiverse/quarkus-mybatis/issues/74 for improving it when using mybatis xml configuration.

Reply:
I did not ask for not creating sqlSessionFactory for each data source. (I have no idea whether that makes sense for Quarkivese MyBatis at all). I asked about the case when the list of sqlSessionFactoryBuildItems is empty. Is it made impossible somewhere in Quarkivese MyBatis or here? Because otherwise `sqlSessionFactoryBuildItem.get(0)` may throw an `ArrayIndexOutOfBoundsException` or similar and it would not inform the end user that (and where) he needs to define a datasource.

Reply:
OK, I get it and I think it should be in Quarkivese MyBatis to make sure to create at least one sqlSessionFactory. Also do you know if it is possible to set ```quarkus.mybatis.xmlconfig.enable=true``` automatically when the ```camel-quarkus-mybatis``` extension is activated ?

Reply:
I think the cleanest way would be to have a dedicated XmlConfigEnabledBuildItem for that in Quarkivese MyBatis. Other options I can think of are hacks.

Reply:
Thanks @ppalaga and I will try this idea in https://github.com/quarkiverse/quarkus-mybatis/issues/84